### PR TITLE
feat: bump zwave-js@9 and @zwave-js/server@1.16.0 and fix breaking changes (#2294)

### PR DESCRIPTION
**Pull request by robertsLando**
cc @AlCalzone

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -878,7 +879,7 @@ app.get(
 			scales: scales,
 		}
 
-		let ports: SerialPort.PortInfo[]
+		let ports: PortInfo[]
```
Comment (AlCalzone) [inline app.ts:882]: You could avoid having to import `@serialport/bindings-cpp` at all if you did this:
```js
try {
  data.serialports = (await SerialPort.list()).map((port) => port.path);
} catch (error) {
  logger.error(error)
  data.serialports = [];
}

res.json(data)
```

File: package.json [hunk 2]
```diff
@@ -92,8 +92,6 @@
     }
   },
   "resolutions": {
-    "serialport": "~9.0.7",
-    "@serialport/bindings": "~9.0.8",
     "colors": "1.4.0"
```
Comment (AlCalzone) [inline package.json:95]: AFAIK npm has reverted the changes to colors, so you can remove this too

File: docker/Dockerfile.contrib [hunk 3]
```diff
@@ -41,9 +41,10 @@ ENV YARN_HTTP_TIMEOUT=300000
 RUN yarn
 RUN yarn build
 
+RUN yarn plugin import version && yarn plugin import workspace-tools
+
 # Update the version info to match the branch built
-RUN yarn lerna version --allow-branch "*" --no-git-tag-version --no-push --ignore-scripts --yes \
-    $(echo $(yarn node -p 'require("semver").inc(require("zwave-js/package.json").version, "prerelease")+".dev"')-$(git rev-parse --short HEAD))
+RUN yarn workspaces foreach version $(echo $(yarn node -p 'require("semver").inc(require("zwave-js/package.json").version, "prerelease")+".dev"')-$(git rev-parse --short HEAD)) --deferred && yarn version apply --all
```
Comment (AlCalzone) [inline docker/Dockerfile.contrib:48]: Does this work now?

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: src/components/custom/StatisticsArrows.vue [hunk 4]
```diff
@@ -48,7 +48,16 @@ export default {
...
 	methods: {
-		jsonToList,
+		jsonToList(item) {
+			item.indented = {
+				a: 'c',
+				b: 'd',
+			}
+			return jsonToList(item, {
+				rtt: 'ms',
+				rssi: 'dBm',
```
[the remark this answers is not in the document]
Comment (AlCalzone) [reply src/components/custom/StatisticsArrows.vue:58]: New v9 test build coming up, this time with `rssiToString(...)` exported from `zwave-js` and `zwave-js/Utils`

File: docker/Dockerfile.contrib (hunk 3 above)
Comment (robertsLando) [reply docker/Dockerfile.contrib:48]: Yep, at least the build completes successfully

**Review by AlCalzone — APPROVED** [review]

**Comment by AlCalzone** [comment]
Bump to beta 6 please

**Comment by raman325** [comment]
Published zwave-js-server 1.16.0-beta.1 which supports zwave-js 9 as well

**Comment by robertsLando** [comment]
@AlCalzone @raman325 Done :) 

@AlCalzone Now I also display sdkVersion in table firmware version

File: src/components/nodes-table/index.vue [hunk 5]
```diff
@@ -240,7 +240,12 @@
 						item.firmwareUpdate.progress
 					}}</span>
 				</v-progress-circular>
-				<div v-else>v{{ item.firmwareVersion }}</div>
+				<div v-else>
+					v{{
+						item.firmwareVersion +
+						(item.sdkVersion ? ` (${item.sdkVersion})` : '')
```
Comment (AlCalzone) [inline src/components/nodes-table/index.vue:246]: 1.: `v` before the firmware version, but no `v` before the SDK version?
2.: I would clearly indicate which is which. firmware version and sdk version have no correlation on 500 series sticks. They do on 700 series sticks with stock firmware, but for Razberry I'd expect them to be different too - they compile their own.

How about:
```
FW: vA.B
SDK: vX.Y.Z
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]

**Comment by kpine** [comment]
I'm testing this build (container metadata says `"org.opencontainers.image.revision": "4969aaeb9d58843396fd0c519a510b203526c545"`), but I'm not seeing any SDK version reflected in the Version column.

image

I also notice there's no `sdkVersion` entry in the node's Debug JSON. Should that be there? The driver logs show it being detected for the controller. I don't see anything in the driver cache either (besides `firmwareVersions`), not sure if it would be stored there anyways?

```
2022-03-17T21:14:09.203Z CNTRLR   received protocol version info:
                                    protocol type:             Z-Wave
                                    protocol version:          7.17.2
                                    appl. framework build no.: 406
                                    git commit hash:           30303030303030303030303030303030
```

File: docker/Dockerfile.contrib [hunk 6]
```diff
@@ -16,7 +16,7 @@ ARG Z2M_REPOSITORY=https://github.com/zwave-js/zwavejs2mqtt
 USER node
 WORKDIR /home/node
 
-RUN git clone -b ${ZWJ_BRANCH} --depth 1 ${ZWJ_REPOSITORY}
+RUN git clone -b ${ZWJ_BRANCH} ${ZWJ_REPOSITORY}
```
[the remark this answers is not in the document]
Comment (robertsLando) [reply docker/Dockerfile.contrib:19]: Couldn't you use `npm pkg get version`?

[the remark this answers is not in the document]
Comment (AlCalzone) [reply docker/Dockerfile.contrib:19]: `yarn version` is used to update the versions in all `package.json`. There's probably a hacky way to do it with `jq` instead if you don't want to clone all history.

**Comment by robertsLando** [comment]
Answering to @kpine 

@AlCalzone Could it be that sdkVersion value comes after node-ready event?

File: docker/Dockerfile.contrib (hunk 6 above)
[the remark this answers is not in the document]
Comment (robertsLando) [reply docker/Dockerfile.contrib:19]: 👍🏼

**Comment by AlCalzone** [comment]
Its queried during the controller interview which the driver does immediately after soft-reset/startup. The value isn't stored anywhere except the controller class instance, so it doesn't appear in the cache.

**Comment by AlCalzone** [comment]
@robertsLando could it by that you're trying to read the `sdkVersion` from the controller node instead of `driver.controller`? There is no special handling for that yet, I forgot to add it.

**Comment by robertsLando** [comment]
> could it by that you're trying to read the sdkVersion from the controller node instead of driver.controller? There is no special handling for that yet, I forgot to add it.

It's that for sure, I'm not reading sdkVersion from driver.controller anywhere as I saw you added it also to nodes. I think it would make sense to add it to controller node too for consistency

**Comment by AlCalzone** [comment]
beta.7 coming up in a few minutes.

**Comment by robertsLando** [comment]
Bumped

**Comment by robertsLando** [comment]
Docker releases of this branch availabe with `zwavejs-9` branch

**Pull request merged**